### PR TITLE
Add Feed RecyclerView with LoadingView

### LINKED ISSUE #12: [UI] Create Feed Layout with RecyclerView

### PR DESCRIPTION
**Pull request by cowboymalboro1884**

### CODE REVIEW

**Comment by cowboymalboro1884** [comment]
https://github.com/GoodTripProject/GoodTrip/assets/112244451/d9d4a536-50ca-47e4-af29-0ac9ea64e850

File: app/src/main/java/ru/hse/goodtrip/model/PostTrip.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package ru.hse.goodtrip.model;
+
+import java.util.ArrayList;
+
+/**
+ * Post data class.
+ */
+public class PostTrip {
+
+  public String timeOfPublication;
```
Comment (shershen0) [inline app/src/main/java/ru/hse/goodtrip/model/PostTrip.java:10]: dont make public fields, it is very bad, instead use getters and setters: lombok @Setters and @Getters for class level annotation

File: app/src/main/java/ru/hse/goodtrip/model/Trip.java [hunk 2]
```diff
@@ -1,4 +1,13 @@
 package ru.hse.goodtrip.model;
 
+import java.util.ArrayList;
+
 public class Trip {
+
+  public String photo;
```
Comment (shershen0) [inline app/src/main/java/ru/hse/goodtrip/model/Trip.java]: same here

File: app/src/main/java/ru/hse/goodtrip/ui/feed/FeedAdapter.java [hunk 3]
```diff
@@ -1,46 +1,121 @@
...
+import ru.hse.goodtrip.ui.feed.FeedViewHolders.FeedPostViewHolder;
 
-public class FeedAdapter extends RecyclerView.Adapter<FeedAdapter.FeedHolder> {
+public class FeedAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>
+    implements View.OnClickListener {
 
-  List<Trip> trips = Collections.emptyList();
+  private static final int VIEW_TYPE_ITEM = 0;
+  private static final int VIEW_TYPE_LOADING = 1;
+  List<PostTrip> items = Collections.emptyList();
```
Comment (shershen0) [inline app/src/main/java/ru/hse/goodtrip/ui/feed/FeedAdapter.java:26]: should be private visibility, its rarely when we want package-private

File: app/src/main/java/ru/hse/goodtrip/ui/feed/FeedViewHolders.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * Main FeedViewHolders class used as namespace for ViewHolders definition.
+ */
+public class FeedViewHolders {
+
+  /**
+   * Creates LoadingViewHolder to appearing when new data is uploading in FeedAdapter.
+   */
+  public static class FeedLoadingViewHolder extends RecyclerView.ViewHolder {
+
+    FeedLoadingViewBinding binding;
```
Comment (shershen0) [inline app/src/main/java/ru/hse/goodtrip/ui/feed/FeedViewHolders.java]: private

File: app/src/main/java/ru/hse/goodtrip/ui/feed/FeedViewHolders.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+import ru.hse.goodtrip.databinding.FeedLoadingViewBinding;
+import ru.hse.goodtrip.databinding.PostTripBinding;
+
+/**
+ * Main FeedViewHolders class used as namespace for ViewHolders definition.
+ */
+public class FeedViewHolders {
```
Comment (shershen0) [inline app/src/main/java/ru/hse/goodtrip/ui/feed/FeedViewHolders.java:11]: nice job! good idea to keep simple holders like that

**Pull request merged**